src/tools/rbd/Utils.cc

```diff
@@ -665,6 +665,21 @@ int init_io_ctx(librados::Rados &rados, const std::string &pool_name,
     return r;
   }
 
+  if (!namespace_name.empty()) {
+    librbd::RBD rbd;
+    std::vector<std::string> names;
+    r = rbd.namespace_list(*io_ctx, &names);
```

Comment:
For security purposes, the ```namespace_list``` cls method might be restricted to a user. Therefore, I think we would actually need a new librbd API method to assert that the namespace exists (which would invoke ```librbd::cls_client::dir_state_assert```)